pkg/apis/node.harvesterhci.io/v1beta1/cloudinit.go

```diff
@@ -0,0 +1,41 @@
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:shortName=nci,scope=Cluster
+// +kubebuilder:subresource:status
+
+type CloudInit struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              CloudInitSpec   `json:"spec"`
+	Status            CloudInitStatus `json:"status,omitempty"`
+}
+
+type CloudInitSpec struct {
+	MatchSelector map[string]string `json:"matchSelector"`
+	Filename      string            `json:"filename"`
+	Contents      string            `json:"contents"`
+}
+
+type CloudInitStatus struct {
+	Rollouts []Rollout `json:"rollouts,omitempty"`
```

Comment:
looking at the way rollouts are evaluated in the controller would it be better to make `Rollouts` a `map[nodeName]RollOut`? 

since the node-manager is going to run as a daemonset, it makes it easier for each controller instance to just lookup Rollout status using its node name and perform updates to only the element in the map?

Reply:
Absolutely, will update.